crates/nix_rs/src/store.rs

```diff
@@ -188,19 +191,15 @@ impl NixStoreCmd {
         }
     }
 
-    /// Given a derivation path, this function recursively queries and return all
+    /// Given the derivation paths, this function recursively queries and return all
     /// of its dependencies in the Nix store.
     pub async fn nix_store_query_requisites_with_outputs(
         &self,
-        drv_path: PathBuf,
-    ) -> Result<Vec<StorePath>, NixStoreCmdError> {
+        drv_paths: &[PathBuf],
+    ) -> Result<HashSet<StorePath>, NixStoreCmdError> {
```

Comment:
This should continue to return `Vec` (so as to maintain original order).

Also, I see no reason for it to require a HashSet. The convenience of a single caller should not universally dictate API of some core function.